### PR TITLE
Remove CustomConfiguration classes from Reverse Engineering

### PR DESCRIPTION
**Pull request by smitpatel**
Rewrite code writers to work directly from Model

### CODE REVIEW

File: src/EFCore.Design/Scaffolding/Internal/CodeWriter.cs [hunk 1]
```diff
@@ -114,9 +113,12 @@ protected CodeWriter([NotNull] IFileService fileService)
         ///     directly from your code. This API may change or be removed in future releases.
         /// </summary>
         public abstract Task<ReverseEngineerFiles> WriteCodeAsync(
-            [NotNull] ModelConfiguration modelConfiguration,
+            [NotNull] IModel model,
             [NotNull] string outputPath,
-            [NotNull] string dbContextClassName,
+            [NotNull] string @namespace,
+            [NotNull] string contextName,
+            string connectionString,
```
Comment (smitpatel) [inline src/EFCore.Design/Scaffolding/Internal/CodeWriter.cs]: missing `NotNull` annotation

File: src/EFCore.Design/Scaffolding/Internal/DbContextWriter.cs [hunk 2]
```diff
@@ -19,80 +22,98 @@ public class DbContextWriter
...
+            {
+                _sb.AppendLine();
+            }
+        }
+
+        private void GenerateEntityTypeErrors(IModel model)
+        {
+            foreach (var entityTypeError in model.Scaffolding().EntityTypeErrors)
+            {
+                _sb.AppendLine($"// {entityTypeError.Value} Please see the warning messages.");
```
Comment (bricelam) [inline src/EFCore.Design/Scaffolding/Internal/DbContextWriter.cs]: "Please see the warning messages" should be a resource. (File a general bug to address)

File: test/EFCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/AllFluentApi/SqlServerReverseEngineerTestE2EContext.expected [hunk 3]
```diff
@@ -303,8 +303,7 @@ namespace E2ETest.Namespace
 
             modelBuilder.Entity<OneToManyDependent>(entity =>
             {
-                entity.HasKey(e => new { e.OneToManyDependentId1, e.OneToManyDependentId2 })
-                    .HasName("PK_OneToManyDependent");
+                entity.HasKey(e => new { e.OneToManyDependentId1, e.OneToManyDependentId2 });
```
Comment (bricelam) [inline test/EFCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/AllFluentApi/SqlServerReverseEngineerTestE2EContext.expected]: 🎉

File: test/EFCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/Attributes/AllDataTypes.expected [hunk 4]
```diff
@@ -68,10 +68,10 @@ namespace E2ETest.Namespace
         [Column("ntextColumn", TypeName = "ntext")]
         public string NtextColumn { get; set; }
         [Column("nvarcharColumn")]
-        [MaxLength(1)]
+        [StringLength(1)]
```
Comment (bricelam) [inline test/EFCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/Attributes/AllDataTypes.expected]: 🎉

File: test/EFCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/Attributes/AttributesContext.expected [hunk 5]
```diff
@@ -43,6 +43,22 @@ namespace E2ETest.Namespace
             modelBuilder.Entity<MultipleFksDependent>(entity =>
             {
                 entity.Property(e => e.MultipleFksDependentId).ValueGeneratedNever();
+
+                entity.HasOne(d => d.RelationA)
+                    .WithMany(p => p.MultipleFksDependentRelationA)
+                    .HasForeignKey(d => d.RelationAid)
+                    .OnDelete(DeleteBehavior.Restrict)
+                    .HasConstraintName("FK_RelationA");
```
Comment (bricelam) [inline test/EFCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/Attributes/AttributesContext.expected]: Were `OnDelete`, `HasConstraintName`, etc. just getting dropped before?

File: test/EFCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/AllFluentApi/SelfReferencing.expected [hunk 6]
```diff
@@ -5,6 +5,11 @@ namespace E2ETest.Namespace
 {
     public partial class SelfReferencing
     {
+        public SelfReferencing()
+        {
+            InverseSelfReferenceFkNavigation = new HashSet<SelfReferencing>();
```
Comment (bricelam) [inline test/EFCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/AllFluentApi/SelfReferencing.expected]: 🎉

File: test/EFCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/Attributes/AttributesContext.expected (hunk 5 above)
Comment (smitpatel) [inline test/EFCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/Attributes/AttributesContext.expected]: Yes. Earlier, it created fluent API only if targeting AK. in all other cases, it generated annotations which swallowed any kind of custom configuration.

File: src/EFCore.Design/Scaffolding/Internal/DbContextWriter.cs (hunk 2 above)
Comment (smitpatel) [inline src/EFCore.Design/Scaffolding/Internal/DbContextWriter.cs]: #8599

**Pull request merged**